client/src/boot/applyTransforms.js

```diff
@@ -55,6 +55,7 @@ const applyTransforms = () => {
 
           return togglePristineState(field, isPristine);
         });
+        form.mutateField('action_UnsavedChangesIndicator', (field) => ({ ...field, isDirty: form.isDirty() }));
```

Comment:
This sets isDirty prop on the component in a react context based of the form dirty state